config/config.go

```diff
@@ -171,15 +171,36 @@ func newIcingaClient(c *SignaliloConfig, l logr.Logger) (icinga2.Client, error)
 		}
 	}
 
-	client, err := icinga2.New(icinga2.WebClient{
-		URL:               c.IcingaConfig.URL,
-		Username:          c.IcingaConfig.User,
-		Password:          c.IcingaConfig.Password,
-		Debug:             c.IcingaConfig.Debug,
-		DisableKeepAlives: c.IcingaConfig.DisableKeepAlives,
-		TLSConfig:         tlsConfig})
-	if err != nil {
-		return nil, err
+	var client *icinga2.WebClient
+
+	var tmp []string
+
+	for _, ur := range c.IcingaConfig.URL {
+		st := strings.Split(ur, ",")
+		for _, b := range st {
+			tmp = append(tmp, b)
+		}
+	}
+
+	c.IcingaConfig.URL = tmp
+
+	for _, url := range c.IcingaConfig.URL {
+		client, err = icinga2.New(icinga2.WebClient{
+			URL:               url,
+			Username:          c.IcingaConfig.User,
+			Password:          c.IcingaConfig.Password,
+			Debug:             c.IcingaConfig.Debug,
+			DisableKeepAlives: c.IcingaConfig.DisableKeepAlives,
+			TLSConfig:         tlsConfig})
+		if err != nil {
+			return nil, err
+		}
+
+		if err = client.TestIcingaApi(url); err != nil {
+			continue
+		} else {
+			break
+		}
 	}
 	return client, nil
```

Comment:
We need to make sure that we have a working client before returning here. Something like

```suggestion
	for _, url := range c.IcingaConfig.URL {
		client, err = icinga2.New(icinga2.WebClient{
			URL:               url,
			Username:          c.IcingaConfig.User,
			Password:          c.IcingaConfig.Password,
			Debug:             c.IcingaConfig.Debug,
			DisableKeepAlives: c.IcingaConfig.DisableKeepAlives,
			TLSConfig:         tlsConfig})
		if err != nil {
			return nil, err
		}

		if err = client.TestIcingaApi(url); err != nil {
			// clear client if the API url wasn't reachable
			client = nil
			continue
		} else {
			break
		}
	}
	if client == nil {
		return nil, fmt.Errorf("No valid Icinga API URL found")
	}
	return client, nil
```